### ISSUE TITLE
thread terminated, error code 999

### ISSUE THREAD

**Issue by kmtam8** [trexminer/T-Rex#1235]
Been using T-Rex Miner for 6 months now.  After upgrading to release 0.25.12, saw this error twice in the past 2 days:

20220417 17:52:05 TREX: thread terminated, error code 999
20220417 17:52:05 WARN: Miner is going to shutdown...
20220417 17:52:05 Main loop finished. Cleaning up resources...
20220417 17:52:05 ApiServer: stopped listening on 192.168.1.82:4067
20220417 17:52:08 T-Rex finished.

**Comment by kctan100** [comment]
I am facing a similar issue as yours. My driver is 512.15. Rig is 6x3090, 2x3080(LHR).

**Comment by thorflea** [comment]
I had just updated hiveos and it ran fine for 24 hours but suddenly started getting  TREX: thread terminated, error code 999
I tried the previous hiveos version but the problem still exists

0.6-215@220415
5.10.0-hiveos #83
N510.60.02

RTX 3060 Ti × 5
RTX 3060 × 2
RTX 3070 Ti × 1
RX 6700 × 1

**Comment by dedizones** [comment]
> I had just updated hiveos and it ran fine for 24 hours but suddenly started getting TREX: thread terminated, error code 999 I tried the previous hiveos version but the problem still exists
> 
> 0.6-215@220415 5.10.0-hiveos #83 N510.60.02
> 
> RTX 3060 Ti × 5 RTX 3060 × 2 RTX 3070 Ti × 1 RX 6700 × 1

what is your arguments ?

my argument no problem

```
"gpu-init-mode": 2
"dag-build-mode": 2
"api-bind-http": 0
"validate-shares": false
"kernel": 4
"lhr-tune": -1
"lhr-autotune-step-size": 0.1
"lhr-autotune-interval": 30
"lhr-autotune-mode": "full"
```

Try update hiveos:

`sudo apt-get update -y && sudo apt-get dist-upgrade -yuf`

`sudo apt-get install nvidia-settings xserver-xorg-core --allow-change-held-packages -y`

**Comment by kctan100** [comment]
> Try update hiveos:

updating hiveos does not resolve the issue either.

**Comment by thorflea** [comment]
I didn't have any configs but changed mine to yours. What stopped the error was removing all overclocks for GPU#3. I will play the the OC later tonight. I will also try your update suggestions tonight. Right now I have run all night,

![Current_Rate](https://user-images.githubusercontent.com/3661316/163787307-d475d779-5e54-4fb2-b7e8-df4a7f77aa14.jpg)

**Comment by everon1218** [comment]
Hey, same error twice with 12 hours difference between. Ive got a screenshot of the first  error code 999
![2022 04 17 trex error code 999](https://user-images.githubusercontent.com/61322704/163823573-1f22856e-5471-4b11-bc1f-4084bfc6e4dc.png)

Args:
"worker": "%WORKER_NAME%"
"devices": "0, 1, 2"
"lhr-autotune-step-size": "0, 0.1, 0.1"
"lhr-autotune-mode": "full"

Clocks
![image](https://user-images.githubusercontent.com/61322704/163824909-ab63d011-3dde-46d4-8fd0-dae277f5454e.png)

**Comment by dedizones** [comment]
> Hey, same error twice with 12 hours difference between. Ive got a screenshot of the first error code 999 ![2022 04 17 trex error code 999](https://user-images.githubusercontent.com/61322704/163823573-1f22856e-5471-4b11-bc1f-4084bfc6e4dc.png)
> 
> Rtx 3090, rtx 3080ti, rtx 3070ti Args: "worker": "%WORKER_NAME%" "devices": "0, 1, 2" "lhr-autotune-step-size": "0, 0.1, 0.1" "lhr-autotune-mode": "full"
> 
> Clocks ![image](https://user-images.githubusercontent.com/61322704/163824909-ab63d011-3dde-46d4-8fd0-dae277f5454e.png)
> 
> Yesterday, gpu 0 was rtx 3090 - I found a cable related issue - turned off from the miner. Today, gpu0 is 3080ti - found nothing but changed the pcie cable for the riser, then reboot. Now, works almost 2 hours with no issue.
> 
> This guide helped me to not go crazy https://github.com/trexminer/T-Rex/wiki/FAQ#what-are-error-codes-15-and-999

Driver version ?

and use or try basic config

**"gpu-init-mode": 2
"dag-build-mode": 2
"api-bind-http": 0
"validate-shares": false
"kernel": 4
"lhr-tune": -1
"lhr-autotune-step-size": 0.1
"lhr-autotune-interval": 30
"lhr-autotune-mode": "full"**

**Comment by everon1218** [comment]
> > Hey, same error twice with 12 hours difference between. Ive got a screenshot of the first error code 999 ![2022 04 17 trex error code 999](https://user-images.githubusercontent.com/61322704/163823573-1f22856e-5471-4b11-bc1f-4084bfc6e4dc.png)
> > Rtx 3090, rtx 3080ti, rtx 3070ti Args: "worker": "%WORKER_NAME%" "devices": "0, 1, 2" "lhr-autotune-step-size": "0, 0.1, 0.1" "lhr-autotune-mode": "full"
> > Clocks ![image](https://user-images.githubusercontent.com/61322704/163824909-ab63d011-3dde-46d4-8fd0-dae277f5454e.png)
> > Yesterday, gpu 0 was rtx 3090 - I found a cable related issue - turned off from the miner. Today, gpu0 is 3080ti - found nothing but changed the pcie cable for the riser, then reboot. Now, works almost 2 hours with no issue.
> > This guide helped me to not go crazy https://github.com/trexminer/T-Rex/wiki/FAQ#what-are-error-codes-15-and-999
> 
> Driver version ?
> 
> and use or try basic config
> 
> **"gpu-init-mode": 2 "dag-build-mode": 2 "api-bind-http": 0 "validate-shares": false "kernel": 4 "lhr-tune": -1 "lhr-autotune-step-size": 0.1 "lhr-autotune-interval": 30 "lhr-autotune-mode": "full"**

Thanks for answering, will try

Nvidia driver 470.103.01
0.6-215@220415
5.10.0-hiveos #83

**Comment by everon1218** [comment]
In case if matter.
After specified configs in the post above miner works smooth. 

Im testing oc now and there is something to

**Comment by dminer3080** [comment]
So I was having the same issue and I ended up going to lol miner and im not having any issue. Looks like its a trex issue

**Comment by dedizones** [comment]
> In case if matter. After specified configs in the post above miner works smooth.
> 
> Im testing oc now and there is something definitely makes an impression.
> 
> If you set higher oc, the miner sets the unlocker to a target that cannot be reached. Although it never manages to reach the target, the unlocker continues to increase the target

Try OC 1185 / 2200 / 0 (PL) for 3070Ti

**Comment by kmtam8** [comment]
The problem does seem to be related to high LHR tune values, as I haven't seen it happen again after the LHR values were reduced by the autotuner.

**Comment by everon1218** [comment]
> > In case if matter. After specified configs in the post above miner works smooth.
> > Im testing oc now and there is something definitely makes an impression.
> > If you set higher oc, the miner sets the unlocker to a target that cannot be reached. Although it never manages to reach the target, the unlocker continues to increase the target
> 
> Try OC 1185 / 2200 / 0 (PL) for 3070Ti

Thanks a lot. I will try next restart. Currently, the 3070ti its on oc 1110/ 2600/ pl 0 with 63.7mh/s 165-173W
Bought the card last month, maybe its rev.2 - cant exceed mem 2750

**Comment by dedizones** [comment]
![image](https://user-images.githubusercontent.com/58809062/164169650-345c56ac-91b6-4bba-90fc-852f42c19ca3.png)

for I never changed more than that my settings based on non FHR

my 3080 FHR and non FHR are at 1085/1800/0

I prefer to lose a few MHS than my cards knowing that also a lot of my time 70% I do ERGO and the rest of ETC

**Comment by setab001** [comment]
greetings from Poland is given by the solution, overcloking cpu off

**Comment by everon1218** [comment]
> ![image](https://user-images.githubusercontent.com/58809062/164169650-345c56ac-91b6-4bba-90fc-852f42c19ca3.png)
> 
> I use 2000 in memory + 400Mhz = 2400mhz ( P0 State in hiveos OC)
> 
> If you use the P0 state like me, don't forget to remove 400Mhs in your OC ^^
> 
>(pic)
> 
> for I never changed more than that my settings based on non FHR
> 
> my 3080 FHR and non FHR are at 1085/1800/0
> 
> I prefer to lose a few MHS than my cards knowing that also a lot of my time 70% I do ERGO and the rest of ETC

@dedizones Hi, Do you suggest me to force p0 state and to lower clocks in order to unlock the memory clocks? Is that better?
The summer is coming here and my only concerns are the temps, and of course the mh/s :)
But seems like 63-64mh/s are the limits of 3070 ti. What do you think?

**Comment by dedizones** [comment]
no the P0 is optional :D

You need to find the right setting for each card, I'll give you a base, but each card has its own OC

**Comment by eren20xx** [comment]
hey, same issue for me i m using 25.12 driver 512.15 for windows i have 5 cards --2060 --1660s --3070 lhr(new) --3060 ti fhr and a 3060 v1 . yesterday i bought 3070. i updated driver which is 470.05 new 512.15  Error code 999 came up all the time today. I changed all cable risers. I reduced the overclock...I reinstalled windows but nothing changes. still error where am i wrong
? please help

**Comment by kmtam8** [comment]
@eren20xx I don't think this is an equipment issue, as I had not seen this issue before upgrading to release 0.25.12.

**Comment by eren20xx** [comment]
> @eren20xx I don't think this is an equipment issue, as I had not seen this error before upgrading to release 0.25.12.

yeah i used trex 25.9 before not and not error.i try 25.12 today but still not working for my rig @trexminer

**Comment by badjazy** [comment]
Out of nowhere I am having this issue. Not sure why or how I would fix this as It doesn't look to be equipment or the clocks

**Comment by dedizones** [comment]
not equipment or the clocks ! try setting with basic setting

**Comment by eren20xx** [comment]
> not equipment or the clocks ! try setting with basic setting

what do you mean basic setting i try --lhr tune --lhr autotune  etc.

**Comment by dedizones** [comment]
Go Discord: new beta version testing: Discord server: https://discord.gg/gj7jcYf

**Comment by trimbilrepo** [comment]
similar issue with #1245 
I have the same problem with Zotac and Gainward brand. working just fine with colorful brand.

**Comment by pinfloyd** [comment]
Obviously the problem is with the pool you're working with. I had a similar situation. After changing the pool, the error ceased to exist. Good luck.

**Comment by Tlarim3187** [comment]
Mine just started doing this overnight, I tried changing driver, rolling back the image of HiveOS, swapped out risers, moved cards around.  Problem seemed to follow one card.  This rig has been stable for months and months prior to upgrading to 25.15.  I tried different algos as well, problem remained. Tried rolling back Trex as far as 25.8 and it still remained. Don’t believe this is a coincidence.

Short term solution was to remove mem OC for affected card that keeps crashing rig, so far it seems stable again.  This is some sort of bug.

**Comment by Tlarim3187** [comment]
> Obviously the problem is with the pool you're working with. I had a similar situation. After changing the pool, the error ceased to exist. Good luck.

Changed pools did not solve this issue for me

**Comment by ashoklathwal** [comment]
[check this detailed article](https://wealthquint.com/fix-trex-error-code-999-57990/) on TREX error code 999, Hope you found useful
